tests/functional/styleguide.js

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public" + '/n' +
+ "* License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global casper */
+'use strict';
+
+var config = require('../lib/config');
+var helpers = require('../lib/helpers');
+var path = '/styleguide/';
+var url = config.base() + path;
+
+casper.test.begin('Styleguide, Elements: ' + url, 3, function suite(test) {
+    casper.start(url, function() {
+        test.assertHttpStatus(200);
+
+        test.assertResourceExists(/media\/img\/styleguide\/home/);
+        test.assertVisible('#sidebar', 'The sidebar navigation is visible');
+    });
+
+    casper.run(function() {
+        test.done();
+        helpers.done();
+    });
+});
+
+casper.test.begin('Styleguide, Navigation flow: ' + url, 11, function suite(test) {
```

Comment:
Makes sense, I was thinking to do an full navigation flow but, perhaps better to make the tests smaller and more succinct. It does make the entire suite run slower now, because there is a lot of `waiting` involved.

Did you have any particular pages in mind?